packages/components/atoms/f-button/src/components/_tests/Button.test.js

```diff
@@ -120,6 +120,36 @@ describe('Button', () => {
                 })
                     .toThrowError('iconButton is set to have buttonType="invalid_value"');
             });
+
+            it.each([
+                'leading',
+                'trailing',
+                false
+            ])('should not throw an error when "hasIcon" is set to %s', hasIcon => {
+                // Arrange
+                const propsData = {
+                    hasIcon,
+                    isIcon: false
```

Comment:
Just out of interest – what currently happens if `isIcon` is set to true and hasIcon is set to `leading` or `trailing`?

Does an error get thrown, or does the component attempt to append an icon to the icon button?